src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1344,13 +1345,23 @@ public sealed override async Task<HttpResponseMessage> SendAsync(HttpRequestMess
             try
             {
                 // Send headers
-                http2Stream = await SendHeadersAsync(request, cancellationToken).ConfigureAwait(false);
-
-                // Send request body, if any
-                await http2Stream.SendRequestBodyAsync(cancellationToken).ConfigureAwait(false);
+                bool hasExpectContinueHeader = request.Content != null && request.HasHeaders && request.Headers.ExpectContinue == true;
 
-                // Wait for response headers to be read.
-                await http2Stream.ReadResponseHeadersAsync().ConfigureAwait(false);
+                if (!hasExpectContinueHeader)
+                {
+                    // Send request body, if any
+                    http2Stream = await SendHeadersAsync(request, cancellationToken).ConfigureAwait(false);
+                    await http2Stream.SendRequestBodyAsync(cancellationToken).ConfigureAwait(false);
+                    // Wait for response headers to be read.
+                    await http2Stream.ReadResponseHeadersAsync().ConfigureAwait(false);
```

Comment:
The preceding 2 lines pass in the cancellationToken.  But this method doesn't.  Why not?

Reply:
Sending part was fixed by #35118 but ReadResponseHeadersAsync was left out. If we need to plumb cancelation to Read*() functions, we can do that as separate PR. Maybe @stephentoub have some suggestions.

Reply:
A separate PR is fine.  But let's not forget about it for 3.0. We don't want to have gaps in our cancellation logic.